### PR TITLE
Support for AnthropicBedrock and AzureOpenAI

### LINKED ISSUE #224: Add support for `AzureOpenAI`
**Is your feature request related to a problem? Please describe.**
There are models hosted on Azure that don't work out of the box with existing SDKs and require special clients.

**Describe the solution you'd like**
Ideally support for models hosted on Azure while still using the same class.

**Describe alternatives you've considered**
Adding additional classes e.g. `AnthropicAzureCall`, but this isn't ideal.

**Additional context**
Many companies use models hosted on Azure, so this seems worth supporting?

### LINKED ISSUE #225: Add support for `AnthropicBedrock`
**Is your feature request related to a problem? Please describe.**
There are models hosted on Bedrock that don't work out of the box with existing SDKs and require special clients.

**Describe the solution you'd like**
Ideally support for models hosted on Bedrock while still using the same class.

**Describe alternatives you've considered**
Adding additional classes e.g. `AnthropicBedrockCall`, but this isn't ideal.

**Additional context**
Many companies use models hosted on Bedrock, so this seems worth supporting?

### PR DESCRIPTION
**Pull request by willbakst**
Closes #224 
Closes #225 

- Add `bedrock_client_wrapper` to `anthropic` module for wrapping `Anthropic` and `AsyncAnthropic` to use AWS Bedrock.
- Add `azure_client_wrapper` to `openai` module for wrapping `OpenAI` and `AsyncOpenAI` to use Microsoft Azure deployments.

### CODE REVIEW

File: docs/concepts/supported_llm_providers.md [hunk 1]
```diff
@@ -367,6 +367,76 @@ class RecipeRecommender(OpenAICall):
     call_params = OpenAICallParams(model="mistral")
 ```
 
+## Using models hosted on major cloud providers
+
+Some model providers have their models hosted on major cloud providers such as AWS Bedricj and Microsoft Azure. You can use provided client wrappers to access these models by providing the proper configuration:
```
Comment (brenkao) [inline docs/concepts/supported_llm_providers.md:372]: `AWS Bedricj` lol

File: mirascope/anthropic/utils.py [hunk 2]
```diff
@@ -1,10 +1,41 @@
...
+    base_url: Optional[Union[str, URL]] = None,
+) -> Callable[
+    [Union[Anthropic, AsyncAnthropic]], Union[AnthropicBedrock, AsyncAnthropicBedrock]
+]:
+    """Returns a client wrapper for using Anthropic models on AWS Bedrock."""
+
+    def inner_wrapper(client: Union[Anthropic, AsyncAnthropic]):
+        """Returns matching `AnthropicBedrock` or `AsyncAnthropicBedrock` client."""
+        kwargs = {
+            "aws_secret_key": aws_secret_key,
```
Comment (brenkao) [inline mirascope/anthropic/utils.py:24]: if user sets in their `.env` file, do they also need to pass it in here? if so, does it make sense for us to get env if it exists?

**Review by brenkao — APPROVED** [review]
approve after fixing nit

File: docs/concepts/supported_llm_providers.md (hunk 1 above)
Comment (willbakst) [reply docs/concepts/supported_llm_providers.md:372]: Good catch

File: mirascope/anthropic/utils.py (hunk 2 above)
Comment (willbakst) [reply mirascope/anthropic/utils.py:24]: I believe that if set to `None` the underlying provider class will grab it from the environment for us, so no need to duplicate that code.

**Pull request merged**